src/lang/langHelpers.ts

```diff
@@ -118,9 +117,17 @@ export async function executeAstMock({
 
 function handleExecuteError(e: any): ExecutionResult {
   let isInterrupted = false
+
   if (e instanceof KCLError) {
     // Detect if it is a force interrupt error which is not a KCL processing error.
-    if (e.msg.includes(EXECUTE_AST_INTERRUPT_ERROR_STRING)) {
+    if (
```

Comment:
TODO: Need to handle this properly.

Since I can call `engineCommandManager.tearDown()` and there are more error handling scenarios the KCL execution workflow will now have 3 new errors.

I've used bad string messages that are confusing to see if the error handling works.

I need to label these error message better.